quote-book/src/synchronized_quote_book.rs

```diff
@@ -3,27 +3,100 @@
 use crate::{Error as QuoteBookError, Pair, Quote, QuoteBook, QuoteId};
 use mc_blockchain_types::BlockIndex;
 use mc_crypto_ring_signature::KeyImage;
-use mc_ledger_db::Ledger;
+use mc_ledger_db::{Error as LedgerError, Ledger};
 use mc_transaction_extra::SignedContingentInput;
-use std::ops::RangeBounds;
 
+use std::{
+    ops::RangeBounds,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    thread::{Builder as ThreadBuilder, JoinHandle},
+    time::Duration,
+};
+
+use crate::Msg;
+use mc_common::logger::{log, Logger};
+use postage::{broadcast::Sender, prelude::Sink};
 /// A wrapper for a quote book implementation that syncs quotes with the ledger
-#[derive(Clone)]
-pub struct SynchronizedQuoteBook<Q: QuoteBook, L: Ledger + Clone + 'static> {
+pub struct SynchronizedQuoteBook<Q: QuoteBook, L: Ledger + Clone + Sync + 'static> {
     /// Quotebook being synchronized to the ledger by the Synchronized Quotebook
     quote_book: Q,
 
+    /// Ledger
     ledger: L,
+
+    /// Shared state
+    shared_state: Arc<Mutex<DbPollSharedState>>,
+
+    /// Join handle used to wait for the thread to terminate.
+    join_handle: Option<JoinHandle<()>>,
+
+    /// Stop request trigger, used to signal the thread to stop.
+    stop_requested: Arc<AtomicBool>,
 }
 
 impl<Q: QuoteBook, L: Ledger + Clone + Sync + 'static> SynchronizedQuoteBook<Q, L> {
     /// Create a new Synchronized Quotebook
-    pub fn new(quote_book: Q, ledger: L) -> Self {
-        Self { quote_book, ledger }
+    pub fn new(quote_book: Q, ledger: L, msg_bus_tx: Sender<Msg>, logger: Logger) -> Self {
```

Comment:
This shouldn't be aware of `Msg`, it will only ever be sending one message (quote removed). As we discussed over Slack, `Msg` is specific to the server crate. Instead, please pass a callback that gets called when a quote has been removed, and the server will pass a callback that sends to the message bus.